src/ceph-volume/ceph_volume/util/device.py

```diff
@@ -105,6 +108,22 @@ def is_device(self):
             return self.disk_api['TYPE'] == 'device'
         return False
 
+    @property
+    def is_valid(self):
+        def reject_device(item, value, reason):
+            if item in self.sys_api:
```

Comment:
this is fine like this, but there is no need to check if `item` is in `.sys_api`, you could just do it with try/except:

    try:
        if self.sys_api[item] == value:
            self._rejected_reasons.append(reason)
    except KeyError:
        pass

Reply:
yup better way ! Thx.